pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -86,7 +86,7 @@ func (f AccessControlDashboardPermissionFilter) Where() (string, []interface{})
 	folderAction := dashboards.ActionFoldersRead
 	dashboardAction := accesscontrol.ActionDashboardsRead
 	if f.PermissionLevel == models.PERMISSION_EDIT {
-		folderAction = accesscontrol.ActionDashboardsCreate
+		folderAction = dashboards.ActionFoldersCreate
```

Comment:
The dashboard create action was added here on purpose. It it used for listing folders where a user can create dashboards in. I think we the search API need to be more flexible (not using old permissions e.g. "Edit"). 

https://github.com/grafana/grafana/pull/46524 adds support to generate  access control filter for several actions. If we then add a new actions filter to the FindPersistedDashboardsQuery we can pass additional actions we want to filter on.